### ISSUE TITLE
Find in project with directory pattern does not respect Ignored Names below root

### ISSUE THREAD

**Issue by jldec** [atom/find-and-replace#621]
This problem seems to occur when there are  ignored directories which are not at the root E.g. a node_modules directory in a subdirectory.
- ignored names config contains `.git, node_modules`
- using Atom v1.3.1 on OS-X 10.11.1

```
git clone https://github.com/jldec/test-atom-ignored-names.git
cd test-atom-ignored-names
cd test
npm install
cd ..
atom .
```

search for `glob` using `Find in project`
- with nothing in the File/directory pattern field, 3 results are returned
- with `test` in the File/directory pattern field, lots of results from `test/node_modules` are returned

**Comment by madhan5000** [comment]
I can confirm this with Atom 1.3.2 on OS-X 10.11.2

**Comment by fritzwf** [comment]
I found that entering this:
    ignoredNames: [
      ".svn"
      ".svn-base"
    ]

does not work to exclude the files.  But entering this:
    ignoredNames: [
      "*.svn"
      "*.svn-base"
    ]
Does work to exclude files.

**Comment by adambuczynski** [comment]
I can confirm this is still an issue in Atom 1.4.1. Quite strange, because I'm sure it used to work previously. However, now I am getting a lot of search results back from my `node_modules` folder and it's slowing down my searches significantly due to the amount of folders and files in there.

Ideally, I'd like to see a separate `Ignored names` setting per package (optionally merging with the main setting).

**Comment by nicolas-van** [comment]
I also confirm this issue on Atom 1.5.1 on OSX El Capitan.

**Comment by gaggle** [comment]
+1

**Comment by BraidedClimber** [comment]
Confirming this is an issue with v 1.8.0 on El Cap 10.11.5

Irritating enough to switch back to sublime.

**Comment by adambuczynski** [comment]
I think this issue is related: https://github.com/atom/find-and-replace/issues/636

I've posted a [comment](https://github.com/atom/find-and-replace/issues/636#issuecomment-220155663) there describing how to reproduce exactly this behaviour.

**Comment by adambuczynski** [comment]
This issue is spilling over into atom/fuzzy-finder as well, as I suspect they share a cache of found files.

After I did a search which included results from a nested `node_modules` folder erroneously, as per this bug, those files then also started appearing in the fuzzy finder:

![image](https://cloud.githubusercontent.com/assets/490562/17499025/f7b28926-5e1e-11e6-9932-9c8dfaff00de.png)

**Comment by 50Wliu** [comment]
The issues are probably related so fixing this has a good chance of also fixing the fuzzy-finder one.

**Comment by sneakypete81** [comment]
atom/scandal#37 should fix this for the "Ignored Names" case.

**Comment by pbeshai** [comment]
FYI while we wait for this to get updated, you can workaround it by adding `, !node_modules` to your search directory. Annoying to have to write it every time, but better than searching through it.

![image](https://cloud.githubusercontent.com/assets/793847/19014532/5a06e49c-87bd-11e6-820d-807adf402cf1.png)

**Comment by jldec** [comment]
This is fixed.

**Comment by dj-hedgehog** [comment]
When did that happen?

![tgc2a3l](https://cloud.githubusercontent.com/assets/7140800/23028970/38250fde-f469-11e6-86d0-4c526ae82c43.jpg)

**Comment by jldec** [comment]
> When did that happen?

I don't know when it happened - i just tested it and the issue appears to be resolved.

**Comment by sneakypete81** [comment]
This was fixed in atom/scandal#37 back in August. Looks like I forgot to update this bug once it got pulled into Atom.

It appears atom/scandal#40 fixed a similar issue with VCS ignores.
